client/tx/search.go

```diff
@@ -184,3 +198,50 @@ func SearchTxRequestHandlerFn(cliCtx context.CLIContext, cdc *codec.Codec) http.
 		utils.PostProcessResponse(w, cdc, txs, cliCtx.Indent)
 	}
 }
+
+func parseHTTPArgs(r *http.Request) (tags []string, page, limit int, err error) {
+	tags = make([]string, 0, len(r.Form))
+	for key, values := range r.Form {
+		if key == "page" || key == "limit" {
+			continue
+		}
+		var value string
+		value, err = url.QueryUnescape(values[0])
+		if err != nil {
+			return
+		}
+
+		var tag string
+		if key == types.TxHeightKey {
+			tag = fmt.Sprintf("%s=%s", key, value)
+		} else {
+			tag = fmt.Sprintf("%s='%s'", key, value)
+		}
+		tags = append(tags, tag)
+	}
+
+	pageStr := r.FormValue("page")
+	if pageStr == "" {
+		pageStr = "1"
+	}
+	page, err = strconv.Atoi(pageStr)
+	if err != nil {
+		return
+	} else if page <= 0 {
+		err = errors.New("page must greater than 0")
+		return
+	}
+
+	limitStr := r.FormValue("limit")
+	if limitStr == "" {
+		limitStr = "30" // should be consistent with tendermint/tendermint/rpc/core/pipe.go:19
+	}
+	limit, err = strconv.Atoi(limitStr)
+	if err != nil {
+		return
+	} else if limit <= 0 {
+		err = errors.New("limit must greater than 0")
```

Comment:
No need to set the variable, just return `tags, limit, page, errors.New("limit must greater than 0")`